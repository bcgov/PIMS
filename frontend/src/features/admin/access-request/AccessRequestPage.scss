@import '../../../variables';
.accessRequestPage {
  overflow-y: scroll;
  height: calc(100vh - #{$header-height} - #{$navbar-height} - #{$footer-height});
}

.accessRequestPage-header {
  font-size: xx-large;
  margin: 20px 0 25px 0;
}

.userInfo {
  max-width: 500px;
  margin: auto;
}

/*
  The Access Request form consists of rows of user info, 
  this selector adds spacing between the rows so that they are easily readable. 
*/
.userInfo > * > div.row,
div.hstack {
  margin-top: 12.5px;
  text-align: left;
}

/*

*/

.roles-input > div {
  width: 100%;
}

.position-input > div {
  width: 100%;
  padding: 0;
}

.notes-input > div {
  width: 100%;
  padding: 0;
}

.success {
  display: flex;
  background-color: lightgreen;
  margin: auto;
  padding: 5px;
}

.success p {
  text-align: center;
  vertical-align: middle;
  margin: auto;
}

.info {
  display: flex;
  background-color: lightskyblue;
  margin: auto;
  padding: 5px;
}

.info p {
  text-align: center;
  vertical-align: middle;
  margin: auto;
}

.notes {
  padding-bottom: 30px;
}

.isDisabled {
  padding-bottom: 10px;
}

.cancelSave {
  padding-top: 15px;
  padding-bottom: 10px;
}

.navBar {
  font-weight: 700;
  text-align: left;
}

div.required {
  color: black;
  font-weight: normal;
}

.required {
  color: red;
}
