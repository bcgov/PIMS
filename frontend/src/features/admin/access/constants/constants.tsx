import { ColumnWithProps } from 'components/Table';
import { AccessStatusDisplayMapper } from 'constants/accessStatus';
import * as React from 'react';
import { Link } from 'react-router-dom';
import { CellProps } from 'react-table';

import { RowActions } from '../components/RowActions';
import { IAccessRequestModel } from '../interfaces';

export const columnDefinitions: ColumnWithProps<IAccessRequestModel>[] = [
  {
    Header: 'IDIR/BCeID',
    accessor: 'username',
    align: 'left',
    Cell: (props: CellProps<IAccessRequestModel>) => {
      return (
        <Link to={`/admin/user/${props.row.original.userId}`}>{props.row.original.username}</Link>
      );
    },
  },
  {
    Header: 'First name',
    accessor: 'firstName',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Last name',
    accessor: 'lastName',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Email',
    accessor: 'email',
    align: 'left',
    clickable: true,
    minWidth: 200,
  },
  {
    Header: 'Position',
    accessor: 'position',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Status',
    accessor: 'status',
    align: 'left',
    clickable: true,
    width: 100,
    Cell: (props: CellProps<IAccessRequestModel>) =>
      AccessStatusDisplayMapper[props.row.original.status],
  },
  {
    Header: 'Agency',
    accessor: 'agency',
    align: 'left',
    clickable: true,
    minWidth: 200,
  },
  {
    Header: 'Role',
    accessor: 'role',
    align: 'left',
    clickable: true,
    minWidth: 200,
  },
  {
    Header: ' ',
    Cell: RowActions,
    width: 75,
  },
];
