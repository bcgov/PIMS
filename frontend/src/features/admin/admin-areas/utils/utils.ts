import { IAdministrativeArea, IApiAdminArea } from '../interfaces';

/** generate administrative area ready for use in api endpoints */
export const toApiAdminArea = (adminArea: IAdministrativeArea, name?: string): IApiAdminArea => {
  return {
    name: name ?? adminArea.name,
    boundaryType: adminArea.boundaryType ?? undefined,
    abbreviation: adminArea.abbreviation ?? undefined,
    groupName: adminArea.groupName ?? undefined,
    id: adminArea.id,
    isDisabled: adminArea.isDiabled ?? false,
    code: String(adminArea.id),
    rowVersion: adminArea.rowVersion ?? '',
    type: 'AdministrativaArea',
  };
};
