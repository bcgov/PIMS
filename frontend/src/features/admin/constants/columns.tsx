import { ColumnWithProps } from 'components/Table';
import { IAgency, IAgencyRecord } from 'interfaces';
import * as React from 'react';
import { Link } from 'react-router-dom';
import { CellProps } from 'react-table';

export const columnDefinitions: ColumnWithProps<IAgencyRecord>[] = [
  {
    Header: 'Agency name',
    accessor: 'name',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Short name',
    accessor: 'code',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Description',
    accessor: 'description',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Parent Agency',
    accessor: 'parent',
    align: 'left',
    clickable: true,
    Cell: (props: CellProps<IAgency>) => {
      return (
        <Link to={`/admin/agency/${props.row.original.parentId}`}>{props.row.original.parent}</Link>
      );
    },
  },
];

export const adminAreasColumnDefinistions: ColumnWithProps<any>[] = [
  {
    Header: 'Area name',
    accessor: 'name',
    align: 'left',
    clickable: true,
  },
];
