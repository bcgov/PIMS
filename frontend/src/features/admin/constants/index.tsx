import * as React from 'react';
import { ColumnWithProps } from 'components/Table';
import { CellProps } from 'react-table';
import { Link } from 'react-router-dom';
import { IAgency, IAgencyRecord } from 'interfaces';

export const columnDefinitions: ColumnWithProps<IAgencyRecord>[] = [
  {
    Header: 'Agency name',
    accessor: 'name',
    align: 'left',
    sortable: true,
    clickable: true,
  },
  {
    Header: 'Short name',
    accessor: 'code',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Description',
    accessor: 'description',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Parent Agency',
    accessor: 'parent',
    align: 'left',
    clickable: true,
    Cell: (props: CellProps<IAgency>) => {
      return (
        <Link to={`/admin/agency/${props.row.original.parentId}`}>{props.row.original.parent}</Link>
      );
    },
    sortable: true,
  },
];
