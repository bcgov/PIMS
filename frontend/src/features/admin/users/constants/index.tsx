import { ColumnWithProps } from 'components/Table';
import * as React from 'react';
import { FaCheck, FaTimes } from 'react-icons/fa';
import { Link } from 'react-router-dom';
import { CellProps } from 'react-table';

import { RowActions } from '../components/RowActions';
import { IUserRecord } from '../interfaces/IUserRecord';

export const columnDefinitions: ColumnWithProps<IUserRecord>[] = [
  {
    Header: 'IDIR/BCeID',
    accessor: 'username',
    align: 'left',
    Cell: (props: CellProps<IUserRecord>) => {
      return <Link to={`/admin/user/${props.row.original.id}`}>{props.row.original.username}</Link>;
    },
    sortable: true,
  },
  {
    Header: 'First name',
    accessor: 'firstName',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Last name',
    accessor: 'lastName',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Email',
    accessor: 'email',
    align: 'left',
    clickable: true,
    sortable: true,
    minWidth: 200,
  },
  {
    Header: 'Position',
    accessor: 'position',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Active',
    accessor: 'isDisabled',
    align: 'left',
    clickable: true,
    sortable: true,
    width: 100,
    Cell: (props: CellProps<IUserRecord>) =>
      props.row.original.isDisabled ? <FaTimes /> : <FaCheck />,
  },
  {
    Header: 'Agency',
    accessor: 'agency',
    align: 'left',
    sortable: true,
    clickable: true,
    minWidth: 200,
  },
  {
    Header: 'Roles',
    accessor: 'roles',
    align: 'left',
    clickable: true,
    minWidth: 200,
  },
  {
    Header: 'Last Login',
    accessor: 'lastLogin',
    align: 'left',
    clickable: true,
    minWidth: 100,
  },
  {
    Header: 'Created On',
    accessor: 'createdOn',
    align: 'left',
    clickable: true,
    sortable: true,
    minWidth: 100,
  },
  {
    Header: ' ',
    Cell: RowActions,
    width: 75,
  },
];
