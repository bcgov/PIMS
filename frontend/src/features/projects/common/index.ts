export * from './interfaces';
export * from './components/FilterBar';
export * from './projectsActionCreator';
export { default as FilterBar } from './components/FilterBar';
export { default as StepStatusIcon } from './components/StepStatusIcon';
export { default as SresManual } from './components/SresManual';
export { default as StepErrorSummary } from './components/StepErrorSummary';
export { default as ProjectNotes, PrivateNotes, PublicNotes } from './components/ProjectNotes';
export { default as EditButton } from './components/EditButton';
export { default as SelectProjectPropertiesPage } from './components/SelectProjectPropertiesPage';
export { default as FormikTable } from './components/FormikTable';
export { default as ApprovalTransitionPage } from './ApprovalTransitionPage';

export { default as projectSlice } from './slices/projectSlice';
export { default as projectTasksSlice } from './slices/projectTasksSlice';
export { default as projectWorflowSlice } from './slices/projectWorkflowSlice';

export { default as ApprovalConfirmationForm } from './forms/ApprovalConfirmationForm';
export { default as UpdateInfoForm } from './forms/UpdateInfoForm';
export { default as DocumentationForm } from './forms/DocumentationForm';
export { default as ProjectDraftForm } from './forms/ProjectDraftForm';
export { default as AppraisalCheckListForm } from './forms/AppraisalCheckListForm';
export { default as FirstNationsCheckListForm } from './forms/FirstNationsCheckListForm';
export { default as SelectProjectPropertiesForm } from './forms/SelectProjectPropertiesForm';
export { default as TasksForm } from './forms/TasksForm';

export { default as useStepForm } from './hooks/useStepForm';
export { default as useProject } from './hooks/useProject';
export { default as useAgencyResponseTable } from './hooks/useAgencyResponseTable';

export * from './slices/projectSlice';
export * from './slices/projectTasksSlice';
export * from './slices/projectWorkflowSlice';

export * from './strings';

export { default as ProjectRouter } from './ProjectRouter';
export { default as ProjectSummaryView } from './ProjectSummaryView';

export { default as ProjectInformationTab } from './tabs/ProjectInformationTab';
export { default as DocumentationTab } from './tabs/DocumentationTab';
export { default as ErrorTabs } from './tabs/ErrorTabs';
export * from './tabs/tabValidation';
