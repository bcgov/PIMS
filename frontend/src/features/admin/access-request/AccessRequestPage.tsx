import React, { useEffect } from 'react';
import './AccessRequestPage.scss';
import { Container, Row, Col, ButtonToolbar, Button, Alert } from 'react-bootstrap';
import {
  getCurrentAccessRequestAction,
  getSubmitAccessRequestAction,
  toAccessRequest,
} from 'actionCreators/accessRequestActionCreator';
import { IUserInfo, IAccessRequest } from 'interfaces';
import { Formik } from 'formik';
import { Form, Input, TextArea, Select } from '../../../components/common/form';
import { useDispatch, useSelector } from 'react-redux';
import { RootState } from 'reducers/rootReducer';
import { IAccessRequestState } from 'reducers/accessRequestReducer';
import * as API from 'constants/API';
import { DISCLAIMER_URL, PRIVACY_POLICY_URL } from 'constants/strings';
import { AccessRequestSchema } from 'utils/YupSchema';
import useKeycloakWrapper from 'hooks/useKeycloakWrapper';
import { mapLookupCode } from 'utils';
import { AccessRequestStatus } from 'constants/accessStatus';
import { Snackbar, ISnackbarState } from 'components/common/Snackbar';
import useCodeLookups from 'hooks/useLookupCodes';
import TooltipWrapper from 'components/common/TooltipWrapper';
import { AUTHORIZATION_URL } from 'constants/strings';

interface IAccessRequestForm extends IAccessRequest {
  agency: number;
  role: string;
}

/**
 * The AccessRequestPage provides a way to new authenticated users to submit a request
 * that associates them with a specific agency and a role within the agency.
 * If they have an active access request already submitted, it will allow them to update it until it has been approved or disabled.
 * If their prior request was disabled they will then be able to submit a new request.
 */
const AccessRequestPage = () => {
  const keycloakWrapper = useKeycloakWrapper();
  const keycloak = keycloakWrapper.obj;
  const userInfo = keycloak?.userInfo as IUserInfo;
  const dispatch = useDispatch();
  const [alert, setAlert] = React.useState<ISnackbarState>({});

  useEffect(() => {
    dispatch(getCurrentAccessRequestAction());
  }, [dispatch]);

  const data = useSelector<RootState, IAccessRequestState>(
    state => state.accessRequest as IAccessRequestState,
  );

  const { getByType, getPublicByType } = useCodeLookups();
  const agencies = getByType(API.AGENCY_CODE_SET_NAME);
  const roles = getPublicByType(API.ROLE_CODE_SET_NAME);

  const accessRequest = data?.accessRequest;
  const initialValues: IAccessRequestForm = {
    id: accessRequest?.id ?? 0,
    userId: userInfo?.sub,
    user: {
      id: userInfo?.sub,
      username: userInfo?.username,
      displayName: userInfo?.name,
      firstName: userInfo?.firstName,
      lastName: userInfo?.lastName,
      email: userInfo?.email,
      position: accessRequest?.user?.position ?? userInfo?.position ?? '',
    },
    agencies: accessRequest?.agencies ?? [],
    status: accessRequest?.status || AccessRequestStatus.OnHold,
    roles: accessRequest?.roles ?? [],
    note: accessRequest?.note ?? '',
    agency: accessRequest?.agencies?.find(x => x).id,
    role: accessRequest?.roles?.find(x => x).id,
    rowVersion: accessRequest?.rowVersion,
  };

  const selectAgencies = agencies.map(c => mapLookupCode(c, initialValues.agency));
  const selectRoles = roles.map(c => mapLookupCode(c, initialValues.role));

  const checkAgencies = (
    <Select
      label="Agency"
      field="agency"
      required={true}
      options={selectAgencies}
      placeholder={initialValues?.agencies?.length > 0 ? undefined : 'Please Select'}
    />
  );

  const checkRoles = (
    <Form.Group className="check-roles">
      <Form.Label>
        Roles{' '}
        <a target="_blank" rel="noopener noreferrer" href={AUTHORIZATION_URL}>
          Role Descriptions
        </a>
      </Form.Label>
      <TooltipWrapper
        toolTipId="select-roles-tip"
        toolTip="To select multiple roles, hold Ctrl and select options."
      >
        <Select
          label="Role"
          field="role"
          required={true}
          options={selectRoles}
          placeholder={initialValues?.roles?.length > 0 ? undefined : 'Please Select'}
        />
      </TooltipWrapper>
    </Form.Group>
  );

  const button = initialValues.id === 0 ? 'Submit' : 'Update';
  const inProgress = React.useMemo(
    () =>
      initialValues.id !== 0 ? (
        <Alert key={initialValues.id} variant="primary">
          You will receive an email when your request is reviewed.
        </Alert>
      ) : null,
    [initialValues.id],
  );

  return (
    <div className="accessRequestPage">
      <div>
        <h3>Access Request</h3>
      </div>
      <hr />
      <Container fluid={true}>
        <Row className="justify-content-md-center">
          <Formik
            enableReinitialize={true}
            initialValues={initialValues}
            validationSchema={AccessRequestSchema}
            onSubmit={async (values, { setSubmitting }) => {
              try {
                await dispatch(getSubmitAccessRequestAction(toAccessRequest(values)));
                setAlert({
                  variant: 'success',
                  message: 'Your request has been submitted.',
                  show: true,
                });
              } catch (error) {
                setAlert({
                  variant: 'danger',
                  message: 'Failed to submit your access request.',
                  show: true,
                });
              }
              setSubmitting(false);
            }}
          >
            {props => (
              <Form className="userInfo">
                {inProgress}

                <Input
                  label="IDIR/BCeID"
                  field="user.username"
                  placeholder={initialValues.user.username}
                  readOnly={true}
                  type="text"
                />

                <Row>
                  <Col>
                    <Input
                      label="First Name"
                      field="user.firstName"
                      placeholder={initialValues.user.firstName}
                      readOnly={true}
                      type="text"
                    />
                  </Col>
                  <Col>
                    <Input
                      label="Last Name"
                      field="user.lastName"
                      placeholder={initialValues.user.lastName}
                      readOnly={true}
                      type="text"
                    />
                  </Col>
                </Row>

                <Input
                  label="Email"
                  field="user.email"
                  placeholder={initialValues.user.email}
                  readOnly={true}
                  type="email"
                />

                {checkAgencies}

                <Input
                  label="Position"
                  field="user.position"
                  placeholder="e.g) Director, Real Estate and Stakeholder Engagement"
                  type="text"
                />

                {checkRoles}

                <TextArea
                  label="Notes"
                  field="note"
                  placeholder="Please specify why you need access to PIMS and include your manager's name."
                  required={true}
                />

                <p>
                  By clicking request, you agree to our{' '}
                  <a href={DISCLAIMER_URL}>Terms and Conditions</a> and that you have read our{' '}
                  <a href={PRIVACY_POLICY_URL}>Privacy Policy</a>.
                </p>
                {alert.show && (
                  <Snackbar
                    show={alert.show}
                    message={alert.message}
                    variant={alert.variant}
                    onClose={() => setAlert({})}
                  />
                )}
                <Row className="justify-content-md-center">
                  <ButtonToolbar className="cancelSave">
                    <Button className="mr-5" type="submit">
                      {button}
                    </Button>
                  </ButtonToolbar>
                </Row>
              </Form>
            )}
          </Formik>
        </Row>
      </Container>
    </div>
  );
};

export default AccessRequestPage;
