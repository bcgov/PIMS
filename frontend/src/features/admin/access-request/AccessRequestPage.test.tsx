import React from 'react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import { Select } from '../../../components/common/form';
import { Formik } from 'formik';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import AccessRequestPage from './AccessRequestPage';
import { ILookupCode } from '../../../actions/lookupActions';
import { IGenericNetworkAction } from '../../../actions/genericActions';
import { NETWORK } from '../../../constants/reducerTypes';
import * as actionTypes from '../../../constants/actionTypes';
import * as API from 'constants/API';
import * as reducerTypes from 'constants/reducerTypes';
import { render, fireEvent, wait } from '@testing-library/react';
import { fillInput } from 'utils/testUtils';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

Enzyme.configure({ adapter: new Adapter() });

const lCodes = {
  lookupCodes: [
    { name: 'One', id: '1', isDisabled: false, type: 'core operational' },
    { name: 'agencyVal', id: '1', isDisabled: false, type: API.AGENCY_CODE_SET_NAME },
    { name: 'disabledAgency', id: '2', isDisabled: true, type: API.AGENCY_CODE_SET_NAME },
    { name: 'roleVal', id: '1', isDisabled: false, type: API.ROLE_CODE_SET_NAME },
    { name: 'disabledRole', id: '2', isDisabled: true, type: API.ROLE_CODE_SET_NAME },
    {
      name: 'privateRole',
      id: '2',
      isDisabled: false,
      isPublic: false,
      type: API.ROLE_CODE_SET_NAME,
    },
  ] as ILookupCode[],
};

const requestAccess = {
  status: 200,
  isFetching: false,
} as IGenericNetworkAction;

const mockStore = configureMockStore([thunk]);
const history = createMemoryHistory();
const mockAxios = new MockAdapter(axios);
mockAxios.onAny().reply(200, {});

// Simulating a succesful submit
const successStore = mockStore({
  [reducerTypes.LOOKUP_CODE]: lCodes,
  [NETWORK]: {
    [actionTypes.ADD_REQUEST_ACCESS]: requestAccess,
  },
});

// Store without status of 200
const store = mockStore({
  [reducerTypes.LOOKUP_CODE]: lCodes,
  [NETWORK]: {
    addRequestAccess: requestAccess,
  },
});
describe('AccessRequestPage functionality', () => {
  it('renders RequestAccessPage correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router history={history}>
            <AccessRequestPage />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('component functionality when requestAccess status is 200 and fetching is false', () => {
    const componentRender = mount(
      <Provider store={successStore}>
        <Router history={history}>
          <AccessRequestPage />
        </Router>
      </Provider>,
    );

    const testRender = () =>
      render(
        <Provider store={successStore}>
          <Router history={history}>
            <AccessRequestPage />
          </Router>
        </Provider>,
      );

    it('renders dropdown for agenices and roles', () => {
      expect(componentRender.find(Select).length).toBe(2);
    });

    it('initializes form with null for agencies and roles', () => {
      expect(
        componentRender
          .find(Formik)
          .first()
          .prop('initialValues'),
      ).toEqual({
        agencies: [],
        agency: undefined,
        id: 0,
        status: 'OnHold',
        note: '',
        role: undefined,
        roles: [],
        rowVersion: undefined,
        user: {
          displayName: undefined,
          email: undefined,
          firstName: undefined,
          id: undefined,
          lastName: undefined,
          position: '',
          username: undefined,
        },
        userId: undefined,
      });
    });

    it('displays enabled agencies', () => {
      const { queryByText } = testRender();
      expect(queryByText('agencyVal')).toBeVisible();
    });

    it('Does not display disabled agencies', () => {
      const { queryByText } = testRender();
      expect(queryByText('disabledAgency')).toBeNull();
    });

    it('displays enabled roles', () => {
      const { queryByText } = testRender();
      expect(queryByText('roleVal')).toBeVisible();
    });

    it('Does not display disabled roles', () => {
      const { queryByText } = testRender();
      expect(queryByText('disabledRole')).toBeNull();
    });

    it('Does not display private roles', () => {
      const { queryByText } = testRender();
      expect(queryByText('privateRole')).toBeNull();
    });

    it('displays a success message', async () => {
      const { container, getByText } = testRender();
      await fillInput(container, 'agency', '1', 'select');
      await fillInput(container, 'role', '1', 'select');
      await fillInput(container, 'note', 'some notes', 'textarea');
      const submit = getByText('Submit');
      wait(() => {
        fireEvent.click(submit);
        expect(getByText('Your request has been submitted.')).toBeVisible();
      });
    });
  });

  it('does not show success message by default', () => {
    const component = mount(
      <Provider store={store}>
        <Router history={history}>
          <AccessRequestPage />
        </Router>
      </Provider>,
    );
    expect(component.find('div.alert').isEmpty).toBeTruthy();
  });
});
