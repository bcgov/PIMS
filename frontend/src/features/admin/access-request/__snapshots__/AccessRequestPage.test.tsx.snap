// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessRequestPage functionality renders RequestAccessPage correctly 1`] = `
<div
  className="accessRequestPage"
>
  <div>
    <h3>
      Access Request
    </h3>
  </div>
  <hr />
  <div
    className="container-fluid"
  >
    <div
      className="justify-content-md-center row"
    >
      <form
        className="userInfo"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="input-user.username"
          >
            IDIR/BCeID
          </label>
          <input
            className="form-control"
            id="input-user.username"
            name="user.username"
            onBlur={[Function]}
            onChange={[Function]}
            readOnly={true}
            type="text"
          />
        </div>
        <div
          className="row"
        >
          <div
            className="col"
          >
            <div
              className="form-group"
            >
              <label
                className="form-label"
                htmlFor="input-user.firstName"
              >
                First Name
              </label>
              <input
                className="form-control"
                id="input-user.firstName"
                name="user.firstName"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
              />
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="form-group"
            >
              <label
                className="form-label"
                htmlFor="input-user.lastName"
              >
                Last Name
              </label>
              <input
                className="form-control"
                id="input-user.lastName"
                name="user.lastName"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
              />
            </div>
          </div>
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="input-user.email"
          >
            Email
          </label>
          <input
            className="form-control"
            id="input-user.email"
            name="user.email"
            onBlur={[Function]}
            onChange={[Function]}
            readOnly={true}
            type="email"
          />
        </div>
        <div
          className="required form-group"
        >
          <label
            className="form-label"
            htmlFor="input-agency"
          >
            Agency
          </label>
          <select
            className="form-select form-control"
            id="input-agency"
            name="agency"
            onBlur={[Function]}
            onChange={[Function]}
            required={true}
          >
            <option
              value=""
            >
              Please Select
            </option>
            <option
              className="option"
              value="1"
            >
              agencyVal
            </option>
          </select>
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="input-user.position"
          >
            Position
          </label>
          <input
            className="form-control"
            id="input-user.position"
            name="user.position"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="e.g) Director, Real Estate and Stakeholder Engagement"
            type="text"
            value=""
          />
        </div>
        <div
          className="check-roles form-group"
        >
          <label
            className="form-label"
          >
            Roles
             
            <a
              href="https://github.com/bcgov/PIMS/wiki/Architecture-Security"
              rel="noopener noreferrer"
              target="_blank"
            >
              Role Descriptions
            </a>
          </label>
          <div
            className="required form-group"
          >
            <label
              className="form-label"
              htmlFor="input-role"
            >
              Role
            </label>
            <select
              className="form-select form-control"
              id="input-role"
              name="role"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              required={true}
            >
              <option
                value=""
              >
                Please Select
              </option>
              <option
                className="option"
                value="1"
              >
                roleVal
              </option>
            </select>
          </div>
        </div>
        <div
          className="required form-group"
        >
          <label
            className="form-label"
            htmlFor="input-note"
          >
            Notes
          </label>
          <textarea
            className="form-control"
            id="input-note"
            name="note"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Please specify why you need access to PIMS and include your manager's name."
            required={true}
            value=""
          />
        </div>
        <p>
          By clicking request, you agree to our
           
          <a
            href="https://www2.gov.bc.ca//gov/content/home/disclaimer"
          >
            Terms and Conditions
          </a>
           and that you have read our
           
          <a
            href="http://www.gov.bc.ca/gov/content/home/privacy"
          >
            Privacy Policy
          </a>
          .
        </p>
        <div
          className="justify-content-md-center row"
        >
          <div
            className="cancelSave btn-toolbar"
            role="toolbar"
          >
            <button
              className="mr-5 btn btn-primary"
              disabled={false}
              type="submit"
            >
              Submit
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
